### PR TITLE
Added tests for Azure file storage backend

### PR DESCRIPTION
**Pull request by ma-hartma**
Closing #83

### CODE REVIEW

File: storage/backendtester.go [hunk 1]
```diff
@@ -60,8 +61,14 @@ func (b *BackendTest) NewBackendTest(t *testing.T) {
 }
 
 func (b *BackendTest) LocationTest(t *testing.T) {
-	if b.Backend.Location() != b.URL {
-		t.Errorf("%s: Expected %v, got %v", b.Description, b.URL, b.Backend.Location())
+	// since urls get parsed in BackendFromURL and some protocols use url-encoded parameters,
```
Comment (craftamap) [inline storage/backendtester.go:64]: Does this work without breaking the other tests?

File: storage/azure/azure_file_test.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+		if err != nil {
+			return err
+		}
+	}
+	for _, dir := range content.DirectoryItems {
+		DeleteFolder(tb, p+"/"+dir.Name)
+	}
+
+	_, err = directoryUrl.Delete(context.Background())
```
Comment (muesli) [inline storage/azure/azure_file_test.go:81]: This `err` seems unhandled. We should probably just `return err` in the line below.

File: storage/azure/azure_file_test.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		u.Path = path.Join(u.Path, p+"/"+file.Name)
+
+		_, err := azfile.NewFileURL(u, pipeline).Delete(context.Background())
+
+		if err != nil {
+			return err
+		}
+	}
+	for _, dir := range content.DirectoryItems {
+		DeleteFolder(tb, p+"/"+dir.Name)
```
Comment (muesli) [inline storage/azure/azure_file_test.go:78]: We should check the error `DeleteFolder` may return, and abort if it's not nil.

File: storage/backendtester.go (hunk 1 above)
Comment (muesli) [reply storage/backendtester.go:64]: I think this should be fine... (until the tests will prove me wrong)

**Pull request merged**